guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryView.java

```diff
@@ -0,0 +1,87 @@
+package org.guvnor.structure.client.editors.repository.clone;
+
+import com.github.gwtbootstrap.client.ui.constants.ControlGroupType;
+
+public interface CloneRepositoryView {
+
+    interface Presenter {
+
+        void handleCancelClick();
+
+        void handleCloneClick();
+    }
+
+    void init(Presenter presenter, boolean isOuMandatory);
+
+    void hideForm();
+
+    void showForm();
+
+    void addOrganizationalUnitSelectEntry();
+
+    void addOrganizationalUnit(String item);
+
+    void addOrganizationalUnit(String item, String value);
+
+    String getOrganizationalUnit(int index);
+
+    int getSelectedOrganizationalUnit();
+
+    boolean isGitUrlEmpty();
+
+    boolean isNameEmpty();
+
+    String getGitUrl();
+
+    String getUsername();
+
+    String getPassword();
+
+    String getName();
+
+    void setName(String name);
+
+    void setUrlHelpMandatory();
+
+    void setUrlHelpInvalidFormat();
+
+    void setUrlGroupType(ControlGroupType type);
+
+    void setNameHelpMandatory();
+
+    void setNameGroupType(ControlGroupType type);
+
+    void setOrganizationalUnitHelpMandatory();
+
+    void setOrganizationalUnitGroupType(ControlGroupType type);
+
+    void setNameEnabled(boolean enabled);
+
+    void setOrganizationalUnitEnabled(boolean enabled);
+
+    void setGitUrlEnabled(boolean enabled);
+
+    void setUsernameEnabled(boolean enabled);
+
+    void setPasswordEnabled(boolean enabled);
+
+    void setCloneEnabled(boolean enabled);
+
+    void setCancelEnabled(boolean enabled);
+
+    void setPopupCloseVisible(boolean closeVisible);
+
+    void showBusyPopupMessage();
```

Comment:
You don't call this anywhere.. perhaps you should, before the call to the server to clone a Repository?

The implementation of this *normally* uses ```BusyPopup``` which defers is display until after 250ms.. so you can "show" it safe in the knowing if the server-side operation completes in <250ms the ```BusyPopup``` will not be shown to Users (provided you obviously ```closeBusyPopup```!)